ext/intl/msgformat/msgformat.c

```diff
@@ -74,12 +76,31 @@ static int msgfmt_ctor(INTERNAL_FUNCTION_PARAMETERS)
 	(mfo)->mf_data.orig_format_len = pattern_len;
 
 	/* Create an ICU message formatter. */
-	MSG_FORMAT_OBJECT(mfo) = umsg_open(spattern, spattern_len, locale, NULL, &INTL_DATA_ERROR_CODE(mfo));
+	MSG_FORMAT_OBJECT(mfo) = umsg_open(spattern, spattern_len, locale, &parse_error, &INTL_DATA_ERROR_CODE(mfo));
 
 	if(spattern) {
 		efree(spattern);
 	}
 
+	if (INTL_DATA_ERROR_CODE( mfo ) == U_PATTERN_SYNTAX_ERROR) {
+		char *msg = NULL;
+		smart_str parse_error_str;
+		parse_error_str = intl_parse_error_to_string( &parse_error );
+		spprintf( &msg, 0, "pattern syntax error (%s)", parse_error_str.s? ZSTR_VAL(parse_error_str.s) : "" );
+		smart_str_free( &parse_error_str );
+		intl_error_set_code( NULL, INTL_DATA_ERROR_CODE( mfo ) );
+
+		if( msg != NULL )
```

Comment:
This null check is unnecessary, spprintf always produces a non-NULL string.

Reply:
I'll add a mixup (and rebase on top of 7.4), but FYI, this code is somewhat "heavily inspired" by https://github.com/php/php-src/blob/c3365bb301fd0a0cb6b5e3b6df5f6742d4262dca/ext/intl/transliterator/transliterator_methods.c#L171-L179

I'll make a followup PR to get rid of the null check there too 😝